### PR TITLE
fix: blob conversion in VHD builds using MSI

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

**What this PR does / why we need it**:
fixes recent changes made to use the agent pool's identity for converting built VHDs from gallery image versions to classic storage account blobs

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by djsly — APPROVED** [review]

File: vhdbuilder/packer/convert-sig-to-classic-storage-account-blob.sh [hunk 1]
```diff
@@ -67,8 +68,9 @@ echo "Converted $sig_resource_id to $disk_resource_id"
 # shellcheck disable=SC2102
 sas=$(az disk grant-access --ids $disk_resource_id --duration-in-seconds 3600 --query [accessSas] -o tsv)
 
-echo "Logging into azcopy with MSI"
-azcopy-preview login --identity
+echo "Using azcopy with pool identity: $POOL_IDENTITY_RESOURCE_ID"
+export AZCOPY_AUTO_LOGIN_TYPE="MSI"
```
Comment (AlisonB319) [inline vhdbuilder/packer/convert-sig-to-classic-storage-account-blob.sh:72]: So now we don't need to login at all, azcopy will use the existing creds?

Comment (cameronmeissner) [reply vhdbuilder/packer/convert-sig-to-classic-storage-account-blob.sh:72]: we don't need the login command when delegating to a managed identity: https://github.com/Azure/azure-storage-azcopy/wiki/Improved-login-support-for-AzCopy-commands-(with-in-memory-secret-store)#using-a-managed-identity

Comment (AlisonB319) [reply vhdbuilder/packer/convert-sig-to-classic-storage-account-blob.sh:72]: thats pretty neat!

**Review by AlisonB319 — APPROVED** [review]

File: .pipelines/templates/.builder-release-template.yaml [hunk 2]
```diff
@@ -196,6 +196,7 @@ steps:
         -e LOCATION="${AZURE_LOCATION}" \
         -e OS_TYPE="Linux" \
         -e CLASSIC_BLOB=${CLASSIC_BLOB} \
+        -e POOL_IDENTITY_RESOURCE_ID=$(POOL_IDENTITY_RESOURCE_ID) \
```
Comment (djsly) [inline .pipelines/templates/.builder-release-template.yaml:199]: where is the ENV coming from ? ADO pipeline Var ?

**Pull request merged**